src/Git/PickLastMinorVersionFromCollection.php

```diff
@@ -38,13 +38,9 @@ public function assert(Version $version) : bool
         $lastVersion = $versionsSortedDescending->first();
 
         $matchingMinorVersions = $stableVersions
-            ->matching(new CompositeConstraint(
-                CompositeConstraint::OPERATOR_AND,
-                new ComparisonConstraint(ComparisonConstraint::OPERATOR_LTE, $lastVersion),
-                new ComparisonConstraint(
-                    ComparisonConstraint::OPERATOR_GTE,
-                    Version::fromString($lastVersion->getMajor() . '.' . $lastVersion->getMinor() . '.0')
-                )
+            ->matching(CompositeConstraint::and(
+                OperationConstraint::lessOrEqualTo($lastVersion),
```

Comment:
Static analysis check complains about possible null pointer error for `$lastVersion`. `VersionCollection->first()` indeed returns nullable type but there is an Assert at the beginning of this `forVersions()` method that makes sure that it deals only with non-empty collections.

I guess Psalm is not sophisticated enough to figure that out?

Reply:
I'll consider reverting `first()` and `last()` to raise "collection is empty" exception.

Reply:
If the collection type is now allowing empty collections, it is correct that type analysis raises an issue here. An assertion on the result will be needed

Reply:
It always allowed. The only difference is that I decided to replace `CollectionIsEmpty` exception with nullable return type. Now I'm questioning that decision.